lambdas/tdr-draft-metadata-checks/src/test/scala/uk/gov/nationalarchives/tdr/draftmetadatachecks/LambdaSpec.scala

```diff
@@ -27,58 +26,83 @@ class LambdaSpec extends ExternalServicesSpec {
 
   val consignmentId: Object = "f82af3bf-b742-454c-9771-bfd6c5eae749"
   val mockContext: Context = mock[Context]
+  val pattern = "yyyy-MM-dd"
+  val dateFormat = new SimpleDateFormat(pattern)
 
-  def mockS3GetResponse(fileName: String): StubMapping = {
-    val filePath = getClass.getResource(s"/$fileName").getFile
-    val bytes = Files.readAllBytes(Paths.get(filePath))
-    wiremockS3.stubFor(
-      get(urlEqualTo(s"/$consignmentId/sample.csv"))
-        .willReturn(aResponse().withStatus(200).withBody(bytes))
-    )
-  }
+  private def checkFileError(errorFile: String) = {
+    mockS3ErrorFilePutResponse()
+    val input = Map("consignmentId" -> consignmentId).asJava
+    new Lambda().handleRequest(input, mockContext)
 
-  def mockS3ErrorFilePutResponse(): StubMapping = {
-    wiremockS3.stubFor(
-      put(urlEqualTo(s"/$consignmentId/draft-metadata-errors.json"))
-        .willReturn(aResponse().withStatus(200))
-    )
-  }
+    val s3Interactions: Iterable[ServeEvent] = wiremockS3.getAllServeEvents.asScala.filter(serveEvent => serveEvent.getRequest.getMethod == RequestMethod.PUT).toList
+    s3Interactions.size shouldBe 1
 
-  val pattern = "yyyy-MM-dd"
-  val dateFormat = new SimpleDateFormat(pattern)
+    val errorWriteRequest = s3Interactions.head
+    val rawErrorFileData = errorWriteRequest.getRequest.getBodyAsString
 
-  "handleRequest" should "download the draft metadata csv file, validate, save empty error file to s3 and save metadata to db if it has no errors" in {
-    validateAndSaveMetadata("sample.csv", "json/empty-error-file.json", "Completed")
-  }
+    // Extract just the JSON part from between the first { and last }
+    val errorFileData = rawErrorFileData.substring(
+      rawErrorFileData.indexOf("{"),
+      rawErrorFileData.lastIndexOf("}") + 1
+    )
 
-  "handleRequest" should "download the draft metadata csv file with an empty row, validate, save empty error file to s3 and save metadata to db if it has no errors" in {
-    validateAndSaveMetadata("sample-empty-row.csv", "json/empty-error-file.json", "Completed")
+    val today = dateFormat.format(new Date)
+    val expectedErrorData: String = Source.fromResource(errorFile).getLines.mkString(System.lineSeparator()).replace("$today", today)
+    errorFileData should be(expectedErrorData)
   }
 
-  "handleRequest" should "return 500 response and throw an error message when a call to the api fails" in {
+  private def validateAndSaveMetadata(csvFile: String, errorFile: String, statusValue: String): Unit = {
```

Comment:
I think we make it as test and move it to the `tdr-draftmetadata-persistence` module.